pkg/virt-operator/resource/apply/delete_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package apply
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+var _ = Describe("Deletion", func() {
+
+	Context("CRD deletion", func() {
+		It("Filter needs deletion", func() {
+			crds := []*extv1.CustomResourceDefinition{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "does-not-deletion1",
+						DeletionTimestamp: now(),
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "needs-deletion",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "does-not-deletion2",
+						DeletionTimestamp: now(),
+					},
+				},
+			}
+
+			needDeletion := crdFilterNeedDeletion(crds)
+
+			Expect(len(needDeletion)).To(Equal(1))
```

Comment:
nit: you can go with `Expect(needDeletion).To(HaveLen(1))`, which is more readable IMO.
Applies to the below expectations as well

Reply:
fixed